app/src/main/scala/org/alephium/explorer/persistence/DBRunner.scala

```diff
@@ -42,19 +37,14 @@ object DBRunner {
     }
 
   @SuppressWarnings(Array("org.wartremover.warts.Throw"))
-  def run[R, E <: Effect](databaseConfig: DatabaseConfig[PostgresProfile])(action: DBAction[R, E])(
-      implicit executionContext: ExecutionContext): Future[R] =
-    databaseConfig.db.run(action).recoverWith {
-      case error => Future.failed(DBRunnerFailed(error))
-    }
+  def run[R, E <: Effect](databaseConfig: DatabaseConfig[PostgresProfile])(
+      action: DBAction[R, E]): Future[R] =
+    databaseConfig.db.run(action)
 
   /** Temporary function until all things are made stateless */
   @SuppressWarnings(Array("org.wartremover.warts.Overloading"))
   def run[R, E <: Effect](action: DBAction[R, E])(
-      implicit executionContext: ExecutionContext,
-      databaseConfig: DatabaseConfig[PostgresProfile]): Future[R] =
-    databaseConfig.db.run(action).recoverWith {
-      case error => Future.failed(DBRunnerFailed(error))
-    }
+      implicit databaseConfig: DatabaseConfig[PostgresProfile]): Future[R] =
+    databaseConfig.db.run(action)
```

Comment:
so I guess `db.run` is already failing the future with an Exception?

Reply:
Yep that's true. [Here](https://github.com/slick/slick/blob/main/slick/src/main/scala/slick/basic/BasicBackend.scala#L76) Slick makes sure that `Exception`s fail the `Future`.

Reply:
great :+1: